controllers/configauditreport/configauditreport_controller.go

```diff
@@ -0,0 +1,185 @@
+
+// ConfigAuditReportReconciler reconciles a ConfigAuditReport object
+type ConfigAuditReportReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=aquasecurity.github.io.giantswarm,resources=configauditreports,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=aquasecurity.github.io.giantswarm,resources=configauditreports/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=aquasecurity.github.io.giantswarm,resources=configauditreports/finalizers,verbs=update
+func (r *ConfigAuditReportReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+	_ = r.Log.WithValues("configauditreport", req.NamespacedName)
+
+	report := &aqua.ConfigAuditReport{}
+	if err := r.Client.Get(ctx, req.NamespacedName, report); err != nil {
+		if apierrors.IsNotFound(err) {
+			// Most likely the report was deleted.
+			return ctrl.Result{}, nil
+		}
+
+		// Error reading the object.
+		r.Log.Error(err, "Unable to read configauditreport")
+		return ctrl.Result{}, err
+	}
+
+	if report.DeletionTimestamp.IsZero() {
+		// Give the report our finalizer if it doesn't have one.
+		if !utils.SliceContains(report.GetFinalizers(), ConfigAuditReportFinalizer) {
+			ctrlutil.AddFinalizer(report, ConfigAuditReportFinalizer)
+			if err := r.Update(ctx, report); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+
+		r.Log.Info(fmt.Sprintf("Reconciled %s || Found (D/W/P): %d/%d/%d",
+			req.NamespacedName,
+			report.Report.Summary.DangerCount,
+			report.Report.Summary.WarningCount,
+			report.Report.Summary.PassCount,
+		))
+
+		// Publish summary metrics for this report.
+		publishSummaryMetrics(report)
+
+	} else {
+
+		if utils.SliceContains(report.GetFinalizers(), ConfigAuditReportFinalizer) {
+			// Unfortunately, we can't just clear the series based on one label value,
+			// we have to reconstruct all of the label values to delete the series.
+			// That's the only reason the finalizer is needed at all.
+			r.clearImageMetrics(report)
+
+			ctrlutil.RemoveFinalizer(report, ConfigAuditReportFinalizer)
+			if err := r.Update(ctx, report); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	}
+
+	return defaultRequeue(), nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *ConfigAuditReportReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	err := ctrl.NewControllerManagedBy(mgr).
+		For(&aqua.ConfigAuditReport{}).
+		Complete(r)
+	if err != nil {
+		return errors.Wrap(err, "failed setting up controller with controller manager")
+	}
+
+	return nil
+}
+
+func (r *ConfigAuditReportReconciler) clearImageMetrics(report *aqua.ConfigAuditReport) {
+	// clear summary metrics
+	summaryValues := valuesForReport(report, metricLabels)
+
+	// Delete the series for each severity.
+	for severity := range getCountPerSeverity(report) {
+		v := summaryValues
+		v["severity"] = severity
+
+		// Expose the metric.
```

Comment:
[This was a bad copy-paste](https://github.com/giantswarm/starboard-exporter/blob/2a399de6ae4ae431bc3f7a220c92416502b9c0c4/controllers/vulnerabilityreport_controller.go#L185) on my part. Will fix in a followup.
```suggestion
		// Delete the metric.
```